### DISCUSSION TITLE
Partitioned dependncies

### DISCUSSION THREAD

**Discussion by ghost** [dagster-io/dagster#20714] [Q&A]
I have a daily partitioned asset, and a monthly partitioned asset that depends on the daily partition asset.

Per documentation here: https://docs.dagster.io/concepts/partitions-schedules-sensors/partitioning-assets
the monthly partitioned asset should depend on all daily partitions for that month.

Questions:

- What is the best way to schedule these two assets? Jobs, sensors, auto-materialization?
- What is the default behavior for the monthly partitioned asset, if at the end of the month not all daily partitions are materialized?
- Is there a way to customize the behavior above. E.g., simply fail, retry, or materialize missing daily partitions.

**Comment by jamiedemaria** [comment]
Hi @brandybobandyistaken  the answer to your questions depends a lot on your set up and what you are trying to achieve.

1. Like i mentioned, the answer to this depends on your particular use case. If you have an external event that you want to monitor to trigger materializations of your assets (ie a file appears in a bucket) then sensors would be the right choice. If you want to materialize your assets on a set cadence (ie every day at 9am materialize the previous day for the daily asset, on the 1st of the month at 9am materialize the previous month asset) then schedules will be the right choice. Automaterialization allows you to declaratively specify when you want assets to materialize, so you can write that the daily asset should materialize every day, and the monthly asset should materialize once all upstream data for a month is available. 

2. The behavior will depend on a couple of things, primarily how you are loading the daily data in the monthly asset. If you are using I/O managers, some I/O managers (like gcs and s3) will fail if there are missing partitions, but you can set some metadata to skip this failure and just return the partitions that exist. The database i/o managers (snowflake, bigquery, duckdb, etc) will not raise an error and will return whatever data is available in the time frame

3. There isn't a universal way to configure this, and it will depend on you particular setup. Like i mentioned above, the s3 I/O manager could be configured to raise or not raise and error, but the snowflake i/o manager won't raise an error

**Reply by ghost** [reply]
Thanks for the reply! I should probably mention I'm scheduling airbyte and dbt with dagster.

**daily-nonpartitioned:**
- airbyte populates raw data
- incremental dbt models create/update staging tables from raw airbyte data

**daily-partitioned:**
-  partitioned dbt models create metrics from non-partitioned staging tables

**weekly/monthly partitioned:**
1. weekly/monthly partitioned dbt models aggregate on non-partitioned staging tables
2. weekly/monthly partitioned dbt models aggregate daily partitioned dbt models.


**What I need:**
1. weekly/monthly partitioned dbt models to wait for non-partitioned run on the day to complete ensuring up-to-date data
2. monthly-partitioned dbt assets to wait until all partitions of the daily-partitioned dbt assets have materialized


**Ideas:**
1. run-status sensor on daily-nonpartitioned job that kicks off other jobs that are waiting for up to date data
2. ???

**Reply by jamiedemaria** [reply]
you could definitely construct enough schedules to make all of these assets run on time, but i think for your use case with all these differently partitioned assets, automaterialization could be a good fit. It's at least worth checking out before manually writing all of your schedules!

**Reply by ghost** [reply]
Thanks! I'm going to try auto-materialization. 

### I have one follow-up question:

I have connections to multiple airbyte streams that serve as the root level assets, that then feed into downstream dbt models.

Currently, because I'm loading in dbt assets with one dbt_assets decorator, if any of the upstream airbyte assets fail none of the downstream dbt assets will run.

I've had limited success with creating multiple @dbt_assets decorators, which seems to separate the dbt assets from eachother in terms of upstream dependencies.

What I want to happen, is for the dbt models that don't depend on the failed airbyte sync to continue to run.

Is there any way to do this without loading in dbt models with n @dbt_assets decorators?

I've also posted this here: https://github.com/dagster-io/dagster/discussions/20761

**Reply by jamiedemaria** [reply]
ok cool - we'll respond to your other question in the separate discussion
